src/System.CommandLine/Binding/ModelDescriptor.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace System.CommandLine.Binding
+{
+    public class ModelDescriptor
```

Comment:
AppModels will need to set just some properties. I think this will be a need for some other cases as well. How would you pass the list of PropertyDescriptors to this class?

Reply:
Do you mean set some properties that were not discovered via reflection?